Add created time package list filter, refs #1102

djjuhasz opened the pull request:
Add a package list dropdown filter to find packages created recently (in the last 3 hours to 7 days). I still need to do work to collapse the status tabs into a dropdown list, add the hidden filter panel, etc. to fully implement #1102, but I think this a good point to review the code and functionality.

djjuhasz committed:
Add created time package list filter, refs #1102

Add a package list dropdown filter to find packages created recently
(in the last 3 hours to 7 days). I still need to do work to collapse
the status tabs into a dropdown list, add the hidden filter panel,
etc. to fully implement #1102, but I think this a good point to
review the code and functionality.

djjuhasz commented:
Here's a recording of current "Created" dropdown filter in action:
https://github.com/user-attachments/assets/a0f7ad34-d782-43ce-8932-5d9c5a76c453

sevein commented on dashboard/src/components/__tests__/TimeDropdown.test.ts:
```diff
@@ -0,0 +1,120 @@
+  changeEvent.value = { field, value };
+};
+
+describe("DateDropdown.vue", () => {
```
TimeDropdown.vue

djjuhasz replied:
Oops, missed that one :laughing:

sevein commented on dashboard/src/components/__tests__/TimeDropdown.test.ts:
```diff
@@ -0,0 +1,120 @@
+    expect(button.text()).toBe("Started");
+  });
+
+  it("emits the correct event when a date is selected", async () => {
```
Would either of these be helpful in this specific test? [it.each](https://vitest.dev/api/#test-each) / [it.for](https://vitest.dev/api/#test-for).

sevein commented on dashboard/src/styles/bootstrap.scss:
```diff
@@ -10,23 +10,24 @@
+@import "bootstrap/scss/card";
 @import "bootstrap/scss/close";
-@import "bootstrap/scss/tooltip";
+@import "bootstrap/scss/dropdown";
```
I can't read diffs like this, is this the import added?

djjuhasz replied:
Yes, I added the "bootstrap/scss/dropdown" import.

sevein commented on dashboard/src/components/__tests__/TimeDropdown.test.ts:
```diff
@@ -0,0 +1,120 @@
+import { mount } from "@vue/test-utils";
```
You may have noticed that other component tests in this repo are using `@testing-library/vue` instead. It seemed to be the right choice back when we started this project but I see now that the Vue docs recommend to rely on `@vue/test-utils` when it comes to [component testing](https://vuejs.org/guide/scaling-up/testing#component-testing).

djjuhasz replied:
I didn't notice the difference.  Copilot generated the initial tests with "@vue/test-utils" and I never questioned it.

sevein commented on dashboard/src/pages/packages/index.vue:
```diff
@@ -1,12 +1,7 @@
-import Tabs from "@/components/Tabs.vue";
-import Tooltip from "bootstrap/js/dist/tooltip";
-import UUID from "@/components/UUID.vue";
+// Import vue libs.
```
You may find this eslint plugin useful: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md.

djjuhasz replied:
I've been struggling to get ESLint working for Enduro - my problem right now is it can't resolve the Typescript import paths. :( I'll try to get it working for another hour or two, but after that I think I'll just have to abandon the attempt.

sevein replied:
Makes sense. You could try to see how the Vue project template configures it. I've used that approach before to deal with other issues.

    $ npm create vue@latest

djjuhasz replied:
Okay, I finally got ESLint working with Typescript + Vue and the import plugin. :sweat_smile: 

I've chosen import order rules to organize imports alphabetically and to enforce a newline between import groups — I'm open to changing these choices if the other devs disagree with them.

sevein replied:
Awesome, looks great.

djjuhasz committed:
Add ESLint and the ESLint import plugin

Add the ESLint import plugin to automatically sort Vue.js imports in
the Dashboard code.

After installing ESLint I ran `npx eslint --fix .` to auto-format files
where possible.

sevein reviewed (approved):
LGTM. I've only tested the eslint config.
